R/intervention.R

```diff
@@ -0,0 +1,41 @@
+#' Set up intervention for simulation
+#'
+#' @description
+#' `intvn_scale_r0()` is a helper for the \code{simulation_*} functions. It
+#' modifies the relevant arguments of the offspring distribution in order to
+#' mimic the impact of an intervention. In particular, it scales the mean of
+#' the offspring distribution. Currently, it can only handle the poisson and
+#' negative binomial distributions and errors when other offspring
+#' distributions are specified alongside `intvn_scale_r0`.
+#'
+#' @inheritParams simulate_tree
+#' @param r0_reduction The intervention impact. A scalar between 0 and 1.
+#' Scales the mean of `offspring_dist`. `r0_reduction` = 0 implies
+#' no intervention impact and `r0_reduction` = 1 implies full impact.
+#' @param pars_list Parameter(s) for poisson or negative binomial offspring
+#' distribution.
+#' @return List of the offspring distribution parameter(s) with the mean
+#' scaled by \code{1 - intvn_scale_r0}.
+#' @details
+#' `intvn_scale_r0()` scales the mean of the offspring distribution
+#' by \eqn{1 - r0\_reduction} so that the new mean is given as:
+#' \deqn{(1 - r0\_reduction) \times R_0,} where \eqn{R_0} is the
+#' mean of the poisson and negative binomial distribution.
+#'
+#' @author James M. Azam
+#' @export
+intvn_scale_r0 <- function(r0_reduction, offspring_dist, pars_list) {
```

Comment:
```suggestion
reduce_mean <- function(r0_reduction, offspring_dist, pars_list) {
```
Just a suggestion - I would go with a simpler name that reflects what the function does, not how we interpret it. Similarly could call the argument `mean_reduction` or similar. But it depends on how you envision this to work for other interventions, and if this reflects some kind of common interface/naming scheme.

Reply:
The plan is to use autocomplete for the intervention functions, hence, the prefix "intvn_". I thought "scale_R0" reads more naturally to the user. I was planning on calling the offspring truncation intervention "intvn_truncate_offspring".

Reply:
I have now renamed the function to `intvn_reduce_mean()` and the argument to `intvn_mean_reduction`.